script/audit_deps.py

```diff
@@ -88,26 +84,30 @@ def npm_audit_deps(path, args):
         return 1
 
     # remove the results which match the exceptions
+    if len(ignored_npm_advisories):
+        print('Ignoring NPM advisories ' + ','.join(map(str, ignored_npm_advisories)))
     for i, val in enumerate(result['actions']):
         result['actions'][i]['resolves'] = \
             [d for d in result['actions'][i]['resolves'] if
                 d['id'] not in ignored_npm_advisories]
 
     resolutions, non_dev_exceptions = extract_resolutions(result)
 
-    # Trigger a failure if there are non-dev exceptions
-    if non_dev_exceptions:
+    if resolutions:
         print('Result: Audit finished, vulnerabilities found')
+
+    # Trigger a failure if there are non-dev exceptions
+    if non_dev_exceptions and not args.audit_dev_deps:
```

Comment:
In the case that there are `non_dev_exceptions` and that `args.audit_dev_deps` is set, we seem to be ignoring the non-dev exceptions. Is that intended?

If I'm reading this right then if `--audit_dev_deps` is passed and there are only non-dev exceptions, we end up returning success (`0`).

Reply:
@fmarier i think you may have missed a `not` in the if condition here? if `args.audit_dev_deps` is true, and there are `non_dev_exceptions`, `resolutions` is also non-empty (because it's a superset of non_dev_exceptions) so we should be returning 1 here: https://github.com/brave/brave-core/pull/6661/files#diff-1539a0feacaeda0bff2a27b87da0a165R107

but now that i think about it, this block (where non_dev_exceptions exist and `args.audit_dev_deps` is false) should return 0, not 1. 

to summarize the intended behavior:
* if there are any exceptions and audit_dev_deps is passed, return 1
* if there are dev exceptions only and audit_dev_deps is NOT passed, return 0
* if there are any non-dev exceptions and audit_dev_deps is NOT passed, return 1
* if there are no exceptions, return 0

Reply:
> resolutions is also non-empty (because it's a superset of non_dev_exceptions)

Ah, that's the part I missed, I hadn't realized that `resolutions` contained both.

Reply:
i think so - https://github.com/brave/brave-core/pull/6661/files#diff-1539a0feacaeda0bff2a27b87da0a165R154